src/main/java/Dan/TaskList.java

```diff
@@ -0,0 +1,103 @@
+package dan;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class TaskList {
+    private List<Task> tasks;
+
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public void addTask(String input) throws DanException {
+        String description;
+        String dateString;
+        if (input.startsWith("todo")) {
+            description = input.replace("todo", "").strip();
+            if (description.isEmpty()) {
+                throw new DanException("Please provide me a description for your todo item");
+            }
+            tasks.add(new ToDo(description));
+
+        } else if (input.startsWith("deadline")) {
+            String[] temp = input.replace("deadline","").strip().split("/by");
```

Comment:
Because we are using `temp` in multiple places within this `else if` block, might want to name it more informatively? Perhaps like `inputArr`?

Reply:
Thanks for the suggestion